README.md

```diff
@@ -4,4 +4,217 @@
  - SPDX-License-Identifier: MPL-2.0
  -->
 
-# coffer
+# Coffer
+
+![coffer-logo](./logo.png "Coffer logo")
+
+Coffer is multi-backend password store with multiple frontends. It reaches out to [Vault](https://www.vaultproject.io/), [pass](https://www.passwordstore.org/) and others to store your most precious secrets.
+
+## Installation
+
+??? from releases?
+
+## Usage
+
+### Configuration
+
+First you must configure coffer and tell it how to reach your preferred backend(s). The way you do this depends on which frontend you want to use, please refer to *frontends* for a list. As an example, here is a `config.toml` taken in by the CLI frontend:
+
+```toml
+main_backend = "vault-local"
+
+[[backend]]
+type = "vault-kv"
+name = "vault-local"
+address = "localhost:8200"
+mount = "secret"
+token = "<vault token>"
+
+```
+
+The configuration consists of a single `main_backend` key and N `[[backend]]` sections. Each one must at least have the `type` key, specify which backend you wish to use. The rest depends on each backend, to see a comprehensive list, refer to *backends*.
+
+The coffer CLI will look for its configuration file in the following order:
+1. A file passed to the tool using the `-c/--config` argument
+2. A file passed to the tool using the `COFFER_CONFIG` environment variable
+3. `config.toml` file in the current working directory.
+
+## Actual Usage
+
+### CLI
+
+The CLI is built from a multitude of commands, a quick list can be found below:
+
+```
+coffer view <entry_path> <field?>
+coffer create <entry_path> --tag* <tag> --field* <field_key>=<field_content> \
+	--privatefield* <field_key>=<field_content> --force? --edit?
+coffer set-field <entry_path> <field_key> <field_contents?> \
+	--visibility? <private/public>
+coffer delete-field <entry_path> <field_key>
+coffer find <path> <text> --sort? --filter <???> --filter-field
+coffer rename <old_path> <new_path> --force? --dry-run?
+coffer copy <old_path> <new_path> --force? --dry-run?
+coffer delete <entry_path> --dry-run? --recursive?
+coffer tag <entry_path> <tag_name> --delete?
+
+> entry_path - path to an entry, think of it as a path to a file instead of
+			   a directory
+> path - path to a directory, not an entry
+> field_key - the key of a field, must satisfy this regex: `?regex?`
+> field_value - the value of a field, must satisfy this regex: `?regex?`
+> tag - a tag, assigned to an entry to tag it, must satisfy this regex:
+		`?regex?`
+```
+
+### Web API
+
+Coffer also have a Web API. Currently it runs on `localhost:8081`. In the future we could configure this but at this moment we can't.
+
```

Comment:
Let's also explain how to start the server, i.e. with `stack run coffer-server` or `cabal run coffer-server`

Reply:
Done